develop

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+#
+# install-cli
+#
+# Copy this file into your project to enable easy, guided
+# installation/bootstrapping.
+#
+# Don't like sh/bash/etc? Sure.
+#
+# Love sh/bash/etc? Yeah, but....
+#
+# Let's use it here, to bootstrap whatever tools/libraries/etc. we
+# *really* love for our project.
+#
+# You can name your implementation of this script whatever you like,
+# such as: install
+#
+
+# Update INSTALL_VERSION to require the version of install-cli this
+# script expects
+INSTALL_VERSION=0.0.4
+
+#
+# start bootstrap installation lib
+#
+# This is a *bit* of boilerplate to ensure we've downloaded the correct
+# version of install-cli. (You probably don't need to touch this.)
+#
+
+INSTALL_FILE=.install.${INSTALL_VERSION//./-}.bash.inc
+
+INSTALL_URL=https://raw.githubusercontent.com/dssg/install-cli/$INSTALL_VERSION/install.bash.inc
+
+[ -f $INSTALL_FILE ] || curl -#L $INSTALL_URL -o $INSTALL_FILE
+
+. $INSTALL_FILE
+
+#
+# end bootstrap installation lib
+#
+
+#
+# start project check/install
+#
+
+# pyenv
+
+exists_pyenv() {
+  icli::check_command pyenv
+}
+
+install_pyenv() {
+  curl -#L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+}
+
+require pyenv \
+  exists_pyenv \
+  install_pyenv \
+  --fail-prefix="not found"
```

Comment:
I have been using virtualenv for this project myself. What's the reason behind including environment management in the repository, when it is something external to the project? Do you (DSaPP) have a standardized way of managing environments that you apply to all your projects? What's the implication of adding pyenv for developers that want to use a different solution?

Reply:
Yep – cool.

This is just a development bootstrapping script, which programmatically recommends and automates the installation of standard tools we use here at DSaPP (in this case that's only `pyenv`), _and_ which automates configuration of the project clone for development.

This only adds environment management to the repository in so far as it asks if you would like to install pyenv, *etc.*, and to configure your clone of the project for development. You may use this script to install any or all of its recommendations. If you like, you can use it as simply a checklist of requirements to satisfy on your own.

This does little to your system on the whole, but rather is intended to bootstrap development of this project. For a one-off project with nothing but a pip `requirements.txt`, it might seem like overkill. With a bevy of projects, requiring various versions of Python and variously complex development requirements, (and developers of various levels of experience), this can make a tremendous difference.

If it helps to explain, this is of course using `virtualenv`; we all use `virtualenv` – or should be – or increasingly its now-built-in successor, the `venv` module, invokable as `python -m venv`. This is about more than that: ensuring the right version of Python, tying the project clone to a particular virtualenv (`.python-version` and `pyenv`'s automatic loading of that virtualenv), and the *many* other project requirements which are commonly added to these. (For one, in order to program management of the cloud resources, I'd now like to ensure that `ecs-cli` is installed….)